backend/src/journey/journey.service.ts

```diff
@@ -0,0 +1,77 @@
+import { Inject, Injectable } from "@nestjs/common";
+import {
+  HafasClient,
+  Journeys,
+  JourneysOptions,
+  JourneyWithRealtimeData,
+  Location,
+  LocationsOptions,
+  RefreshJourneyOptions,
+  Station,
+  Stop
+} from "hafas-client";
+import { HAFAS_CLIENT, HAFAS_LANGUAGE } from "../symbols";
+import { JourneyRequest } from "./entities/journey.request.entity";
+
+/** Type alias for results of location query. */
+export type Locations = ReadonlyArray<Station | Stop | Location>;
+
+/**
+ * Service that uses HAFAS to plan journeys.
+ */
+@Injectable()
+export class JourneyService {
+
+  constructor(
+    @Inject(HAFAS_CLIENT) private hafas: HafasClient
+  ) { }
+
+
+  /**
+   * Plans and returns options for the requested journey.
+   *
+   * @param journeyRequest request for a journey
+   * @return different options to plan the journey
+   */
+  planJourney(journeyRequest: JourneyRequest): Promise<Journeys> {
+    const options: JourneysOptions = {
+      results: 3,
```

Comment:
We have to find a good setting here. Should all results be passed to the frontend (would be "results: null") or limit the number of results?